.github/workflows/job_detect_changes.yaml

```diff
@@ -91,9 +104,8 @@ jobs:
 
             # Go services (API v2, Ctrl, Deploy services)
             go:
-              - 'go/**'
-              - '!go/benchmarks/**'
-              - '!go/demo_api/**'
+              - 'go/!(benchmarks|demo_api)/**'
+              - 'go/!(benchmarks|demo_api)'
```

Comment:
Had to change the glob matching because path-filters uses picomatch under the hood, and it has some bug with the way exclusions work. Using extended glob matching or extglob made it work the way we intended because its "match anything in go/ that isn't benchmarks or demo_api" and is more reliable than including-first-then-excluding.